firedrake/slate/slac/compiler.py

```diff
@@ -627,7 +627,7 @@ def gem_to_loopy(gem_expr, var2terminal, scalar_type):
     shape = gem_expr.shape if len(gem_expr.shape) != 0 else (1,)
     idx = make_indices(len(shape))
     indexed_gem_expr = gem.Indexed(gem_expr, idx)
-    args = ([loopy.GlobalArg("output", shape=shape, dtype=scalar_type)]
+    args = ([loopy.GlobalArg("output", shape=shape, dtype=scalar_type, is_output=True, is_input=True)]
```

Comment:
Should TSFC be setting these inout parameters on its globalargs too?

Reply:
I would think so yes, but want so see if there is a test actually failing because it's not there. The test would fail with a KeyError in the Kernel matching.